Sources/Utilities/KMZendeskChatHandler.swift

```diff
@@ -15,41 +15,60 @@ struct KommunicateURL {
     static let dashboardURL = "https://dashboard.kommunicate.io/conversations/"
 }
 
-public class KMZendeskChatHandler {
-
-    public static let shared = KMZendeskChatHandler()
+public class KMZendeskChatHandler : NSObject, JWTAuthenticator {
+  
+    public func getToken(_ completion: @escaping (String?, Error?) -> Void) {
+        completion(jwtToken,nil)
+    }
+    
+   public static let shared = KMZendeskChatHandler()
     var groupId: String = ""
-    var connectionStatus = false
-    var messages:[ALMessage]? = nil
     var zenChatIntialized = false
     var userId = ""
     var isChatTranscriptSent = false
+    var isHandOffHappened = false
+    var lastUserMessageCreatedTime: NSNumber = 0
+    var messageBufffer = [ALMessage]()
+
+    var jwtToken: String = "" {
+            didSet {
+                guard !jwtToken.isEmpty else {
+                    resetIdentity()
+                    return
+                }
+                Chat.instance?.setIdentity(authenticator: self)
+            }
+        }
     
-    public func initiateZendesk(key: String, conversationId: String) {
+    public func initiateZendesk(key: String) {
         Chat.initialize(accountKey: key)
-        groupId = conversationId
         zenChatIntialized = true
         fetchUserId()
         authenticateUser()
-
-        DispatchQueue.main.asyncAfter(deadline: .now() + 2.0) {
-            self.sendChatInfo()
-            self.sendChatTranscript()
-            
-            let connectionToken = Chat.connectionProvider?.observeConnectionStatus { [weak self] (connection) in
-                // Handle connection status changes
-                guard let weakSelf = self,
-                      connection.isConnected,
-                      self?.connectionStatus != true else {
-                   self?.connectToZendeskSocket()
-                   return
-                }
-                weakSelf.connectionStatus = true
-                weakSelf.observeChatLogs()
-                if !weakSelf.isChatTranscriptSent {
-                    weakSelf.sendChatTranscript()
-                }
-           }
+    }
+    
+    public func updateHandoff(_ value: Bool) {
+        self.isHandOffHappened = value
+    }
+    
+    func observeChatLogs() {
```

Comment:
Are these logs or updates?

Reply:
these kind of events like agent message, agent join, agent resolve, agent leave. it comes as Chat Logs. thatswhy named it as `observeChatLogs`